src/napari_graph/interops.py

```diff
@@ -0,0 +1,80 @@
+import warnings
+
+import networkx as nx
+import numpy as np
+import pandas as pd
+
+from napari_graph.base_graph import BaseGraph
+from napari_graph.directed_graph import DirectedGraph
+from napari_graph.undirected_graph import UndirectedGraph
+
+
+def from_networkx(graph: nx.Graph) -> BaseGraph:
+    """
+    Convert a NetworkX graph into a napari-graph UndirectedGraph or DirectedGraph.
+
+    Parameters
+    ----------
+    graph : nx.Graph
+        The NetworkX graph to be converted.
+
+    Returns
+    -------
+    BaseGraph
+        An equivalent napari-graph UndirectedGraph or DirectedGraph.
+    """
+
+    coords_dict = nx.get_node_attributes(graph, "pos")
+    if len(coords_dict) > 0:
+        coords_df = pd.DataFrame.from_dict(coords_dict, orient="index")
+    else:
+        coords_df = None
+
+    edges = np.asarray(graph.edges)
+
+    if edges.ndim > 1 and edges.shape[1] > 2:
+        warnings.warn("Edges weights are not supported yet and were ignored.")
+        edges = edges[:, :2]
```

Comment:
add test?

Reply:
I misinterpreted how networkx edges worked, I noticed this was not an issue when implementing the test and refactored the code.